src/main/java/com/esri/arcgisruntime/toolkit/skins/FeatureTemplatePickerTilePaneSkin.java

```diff
@@ -0,0 +1,329 @@
+    if (swatchesNeedUpdate) {
+      // ensure an update is only triggered once per symbol size change
+      swatchesNeedUpdate = false;
+      // trigger all symbol swatch updates
+      CompletableFuture<?>[] futures = contentPane.getChildren().stream()
+          .flatMap(n -> ((VBox) n).getChildren().stream())
+          .filter(n -> n instanceof TilePane)
+          .flatMap(tilePane -> ((TilePane) tilePane).getChildren().stream())
+          .filter(n -> n instanceof ToggleButton)
+          .map(n -> {
+            ToggleButton toggleButton = (ToggleButton) n;
+            FeatureTemplateItem featureTemplateItem = (FeatureTemplateItem) toggleButton.getUserData();
+            ImageView imageView = (ImageView) toggleButton.getGraphic();
+            return updateSwatch(featureTemplateItem, imageView);
+          })
+          .toArray(CompletableFuture<?>[]::new);
+      // wait until all updates finished before updating layout normally
+      CompletableFuture.allOf(futures).thenRunAsync(() ->
+        super.layoutChildren(contentX, contentY, contentWidth, contentHeight)
+      );
+    } else {
+      super.layoutChildren(contentX, contentY, contentWidth, contentHeight);
+    }
+  }
+
+  /**
+   * Updates the control's content pane and scroll pane based on the control's orientation.
+   */
+  private void updateOrientation() {
+    Orientation orientation = getSkinnable().getOrientation();
+    // keep a reference to the current children
+    List<Node> children = contentPane.getChildren();
+    // create a new content pane based on the orientation
+    contentPane = orientation == Orientation.HORIZONTAL ? new HBox() : new VBox();
+    // add a style class so the picker can be styled via CSS
+    contentPane.getStyleClass().add("feature-template-picker");
+    // add a CSS pseudo class so the picker can be styled based on it orientation
+    contentPane.pseudoClassStateChanged(HORIZONTAL_PSEUDO_CLASS, orientation == Orientation.HORIZONTAL);
+    // copy the children back into the new content pane
+    contentPane.getChildren().addAll(children);
+    scrollPane.setContent(contentPane);
+    // size the content pane to the scroll pane
+    scrollPane.setFitToWidth(orientation == Orientation.VERTICAL);
+    scrollPane.setFitToHeight(orientation == Orientation.HORIZONTAL);
+  }
+
+  /**
+   * Creates a VBox with a label and tile pane representing a feature template group.
+   *
+   * @param featureTemplateGroup feature template group to show
+   * @return VBox
+   */
+  private VBox createVBoxForFeatureTemplateGroup(FeatureTemplateGroup featureTemplateGroup) {
+    VBox vBox = new VBox();
+    // add a style class so the group can be styled via CSS
+    vBox.getStyleClass().add("feature-template-group");
+    vBox.setVisible(false); // wait until the group's feature layer is loaded before displaying
+    vBox.managedProperty().bind(vBox.visibleProperty()); // hide the view when not visible
```

Comment:
Is this intended - setting the managed property? I guess it means the node is not considered in the layout? My experience of unmanned layouts hasn't been great e.g. I think the callout is unmanaged.

In what circumstances do the vboxes get set to not be visible? Can the user access them to do that?

Reply:
Removed.